RTN-023.tex

```diff
@@ -50,6 +50,372 @@
+
+\item Accepts a yaml formatted keyword-value set of parameters which
+defines a single run. (see example run specification yaml below).
+
+\item After 'bps submit <run-spec.yaml>', queries the butler to generate
+a 'quantum graph', a DAG which defines the ordered set of 
+pipeline/pipetask which are to be run on which specific set of input 
+collections (or subsets of a collection which may be picked out via SQL
+type where clauses)
+
+\item Generates an ordered list of tasks, each of which may consist
+of hundreds of quanta, and submits this list submitted to a 
+workload/workflow system (such as HT-Condor or PanDA).
+
+\item Monitors runs which have been submitted  
+	(in the HT-Condor case see Fig. \ref{bpsreport}, currently, could be expanded to PanDA) 
+
+\begin{figure}
+\includegraphics[width=\textwidth]{bpsreport.jpg}]
+\caption{Sample BPS report display showing number of quanta for each task within a give run.
+Credit: M. Gower}
+\label{bpsreport}
+\end{figure}
+
+
+\end{enumerate}
+
+PanDA:
+
+\begin{enumerate}
+
+\item Accepts an ordered list of tasks (from BPS), which specify what
+pipelines to run on which inputs.
+
+\item Distributes tasks to worker nodes, which may be widely distributed.
+
+\item Monitors tasks and associated quanta. See Figs. \ref{pandadash0},\ref{tasks1},\ref{tasks4},\ref{tasks5}
+
+\begin{figure}
+\includegraphics[width=\textwidth]{pandadash0.jpg}]
+\caption{Top level PanDA monitoring dashboard of running and recent jobs
+Credit: S. Padolski}
+\label{pandadash0}
+\end{figure}
+
+\begin{figure}
+\includegraphics[width=\textwidth]{tasks1.jpg}]
+\caption{Next level task list for first job in list
+Credit: H. Lin}
+\label{tasks1}
+\end{figure}
+
+\begin{figure}
+\includegraphics[width=\textwidth]{tasks4.jpg}]
+\caption{Next level status for isr task (2nd from bottom in prev list).
+Credit: H. Lin}
+\label{tasks4}
+\end{figure}
+
+\begin{figure}
+\includegraphics[width=\textwidth]{tasks5.jpg}]
+\caption{Next level status for some of the 603 quanta in this task.
+Credit: H. Lin}
+\label{tasks5}
+\end{figure}
+
+\item Allows viewing of log files for individual tasks. See Fig \ref{log0},\ref{log1}
+
+\begin{figure}
+\includegraphics[width=\textwidth]{log0.jpg}]
+\caption{Getting to the stdout/stderr log for this quantum. 
+Credit: H. Lin}
+\label{log0}
+\end{figure}
+
+\begin{figure}
+\includegraphics[width=\textwidth]{log1.jpg}]
+\caption{A look at stdout/stderr -- if there was an error, one can view it here to help diagnose an issue.
+Credit: H. Lin}
+\label{log1}
+\end{figure}
+
+\item Allows for retries with more memory for individual tasks that fail.
+
+\item More!
+
+\end{enumerate}
+
+Please see \cite{DMTN-168} for more details on running with BPS and PanDA.
+
+\section{Sample BPS 'run' specification yaml}
+
+A 'run' is defined as a run of a BPS-run-spec.yaml through 'bps submit
+BPS-run-spec.yaml'.
+
+This run-spec yaml file defines:
+
+\begin{enumerate}
+
+\item An ordered list of pipetasks to be run: 
```

Comment:
Is "pipetask" here talking about the fact that this is implemented as an execution of the `pipetask` shell command or a statement that quanta are implemented a `PipelineTask` instances? If it's neither of these then it's a bit confusing. Also the plan is to move away from running `pipetask` itself at some point in the future.

Reply:
Perhaps change to "An existing (sub)pipeline definition to use for the run"

Depending upon requests, bps could be changed to have a run be defined as an ordered list of (sub)pipeline definitions by making bps internal steps themselves jobs.  If a predefined sequential list (e.g., step 1, step 2 and step 3 on same inputs), this would be pretty easy to implement in the HTCondor plugin, but would need to discuss whether the other plugin writers forsee any tricky items.  The bigger campaign DAG of (sub)pipelines could also be added to bps, but should look at the other run management tools before coding this large of a feature.